src/test/java/com/training/services/tests/GAAccountTest.java

```diff
@@ -7,55 +7,58 @@
-                header(map.get("ContentTypeHeader"), (map.get("ContentTypeValue")))
-                .body("{\n" +
-                        " \"uid\": \""+uid+"\",\n" +
-                        " \"password\": \"Password1\"\n" +
-                        "}").post("/authentication/login")
-                .then()
-                .extract()
-                .response()
-                .as(Response.class);
+        Map<String, Object> headerMap = new HashMap();
+        headerMap.put(map.get("AppKeyHeader"), map.get("AppKeyValue"));
+        headerMap.put(map.get("ContentTypeHeader"), map.get("ContentTypeValue"));
+        Response authenticationResponse =
+                new RestEngine().getResponsePost(map.get("URI") + "/authentication/login"
+                        , headerMap
+                        , "{\n" +
+                                " \"uid\": \""+uid+"\",\n" +
+                                " \"password\": \"Password1\"\n" +
+                                "}")
+                        .as(Response.class);
         logger.info("status"+authenticationResponse.getStatus());
         logger.info("uid"+authenticationResponse.getPayload().getAccountId());
         logger.info("error"+authenticationResponse.getErrors());
         Assertions.assertEquals(authenticationResponse.getStatus(),"200");
         softAssert.assertThat(authenticationResponse.getErrors().size()).isEqualTo(0);
-        softAssert.assertThat(authenticationResponse.getPayload().getAccountId()).isEqualTo("54d7543e-45f1-4b7b-b83c-fa107f44809b");
+        softAssert.assertThat(authenticationResponse.getPayload().getAccountId()).
+                isEqualTo("54d7543e-45f1-4b7b-b83c-fa107f44809b");
         softAssert.assertThat(authenticationResponse.getPayload().getLoginStatus()).isEqualTo("AUTHENTICATED");
         softAssert.assertThat(authenticationResponse.getPayload().getUid()).isEqualTo(uid);
     }
 
+    /**
+     * testAuthenticateWrongAppKey():post requested with wrong appkey and checked the negative response,
+     * parameters values passed through Json
+     */
     @Test
     public void testAuthenticateWrongAppKey() {
-        RestAssured.baseURI = map.get("URI");
-        Response authenticationResponse= given().
-                header(map.get("AppKeyHeader"), (map.get("AppKeyWrongValue"))).
-                header(map.get("ContentTypeHeader"), (map.get("ContentTypeValue")))
-                .body("{\n" +
-                        " \"uid\": \"testShrikant@api.com\",\n" +
-                        " \"password\": \"Password1\"\n" +
-                        "}").post("/authentication/login")
-                .then()
-                .extract()
-                .response()
-                .as(Response.class);
+        Map<String, Object> headerMap = new HashMap();
+        headerMap.put(map.get("AppKeyHeader"), map.get("AppKeyWrongValue"));
+        headerMap.put(map.get("ContentTypeHeader"), map.get("ContentTypeValue"));
```

Comment:
lets use this in before Each and, directly pass the variable headerMap to RestEngine.